### PR TITLE
Enhancements on .NET Core 3.1

### PR DESCRIPTION
**Pull request by lorenzo-ange**
Hi all,
Here are some little enhancements I'd like to merge to the official runtime image.

### CODE REVIEW

File: stable/dotnetcore/run-webapi.sh [hunk 1]
```diff
@@ -4,4 +4,5 @@
 export ASSEMBLY_NAME="${ASSEMBLY_NAME:=$(basename /kubeless/*.csproj .csproj)}"
 
 cp /kubeless/publish/*.dll /app
+cp /kubeless/publish/*.pdb /app
```
Comment (lorenzo-ange) [inline stable/dotnetcore/run-webapi.sh:7]: keep PDB files to have better exception stacktraces (with line numbers)

File: stable/dotnetcore/src/Kubeless.Core/Handlers/DefaultParameterHandler.cs [hunk 2]
```diff
@@ -37,11 +38,12 @@ private async Task<Event> GetEvent(HttpRequest request)
...
-
-            if (contentType == "application/json" && data.ToString().Length > 0) {
-                data = JsonSerializer.Deserialize<dynamic>(data.ToString());
+            object data;
+            if (contentType.Any(ct => ct.Contains("application/json")) && request.ContentLength > 0) {
+                data = await JsonSerializer.DeserializeAsync<dynamic>(request.Body);
+            } else {
+                using var sr = new StreamReader(request.Body, leaveOpen: true);
+                data = await sr.ReadToEndAsync();
             }
```
Comment (lorenzo-ange) [inline stable/dotnetcore/src/Kubeless.Core/Handlers/DefaultParameterHandler.cs:47]: - use builtin DeserializeAsync method of JsonSerializer 
- fix content-type parsing in some cases (e.g. "Content-Type: application/json; charset=utf-8")

File: stable/dotnetcore/src/Kubeless.WebAPI/Controllers/RuntimeController.cs [hunk 3]
```diff
@@ -32,30 +40,34 @@ public RuntimeController(ILogger<RuntimeController> logger, IInvoker invoker, IP
         [AcceptVerbs("GET", "POST", "PUT", "PATCH", "DELETE")]
         public async Task<object> Execute()
         {
-            _logger.LogInformation("{0}: Function Started. HTTP Method: {1}, Path: {2}.", DateTime.Now.ToString(), Request.Method, Request.Path);
+            _logger.LogInformation($"{DateTime.Now}: Function Started. HTTP Method: {Request.Method}, Path: {Request.Path}.");
```
Comment (lorenzo-ange) [inline stable/dotnetcore/src/Kubeless.WebAPI/Controllers/RuntimeController.cs:44]: refactoring: use string interpolation

File: stable/dotnetcore/src/Kubeless.WebAPI/Controllers/RuntimeController.cs [hunk 4]
```diff
@@ -32,30 +40,34 @@ public RuntimeController(ILogger<RuntimeController> logger, IInvoker invoker, IP
...
 
             Event @event = null;
             Context context = null;
             try
             {
                 (@event, context) = await _parameterHandler.GetFunctionParameters(Request);
 
-                var output = await _invoker.Execute(@event, context);
+                object output;
+                var durationMetrics = DurationSeconds.WithLabels(context.ModuleName, context.FunctionName, context.Runtime);
```
Comment (lorenzo-ange) [inline stable/dotnetcore/src/Kubeless.WebAPI/Controllers/RuntimeController.cs:52]: add duration metric for functions

**Comment by andresmgot** [comment]
cc/ @allantargino in case he is able to do a review

**Pull request merged**